### ISSUE TITLE
arbitrary_source_item_ordering does trigger in impl blocks

### ISSUE THREAD

**Issue by alexkazik** [rust-lang/rust-clippy#17241]
### Summary

The lint does trigger on impl blocks which are ordered according to the trait definition.

### Lint Name

arbitrary_source_item_ordering

### Reproducer

I tried this code:

```rust
impl<'a> IntoIterator for &'a Languages {
    type Item = &'a Language;
    type IntoIter = Iter<'a, Language>;
    fn into_iter(self) -> Self::IntoIter {
        todo!()
    }
}
```

I saw this happen:

```
error: incorrect ordering of items (must be alphabetically ordered)
  --> src/languages.rs:21:10
   |
21 |     type IntoIter = Iter<'a, Language>;
   |          ^^^^^^^^
   |
note: should be placed before `Item`
  --> src/languages.rs:20:10
   |
20 |     type Item = &'a Language;
   |          ^^^^
   = help: for further information visit https://rust-lang.github.io/rust-clippy/master/index.html#arbitrary_source_item_ordering
   = note: `-D clippy::arbitrary-source-item-ordering` implied by `-D warnings`
   = help: to override `-D warnings` add `#[allow(clippy::arbitrary_source_item_ordering)]`
```

I expected to see this happen:
Nothing

### Version

```text
rustc 1.98.0-nightly (f8a08b688 2026-05-30)
binary: rustc
commit-hash: f8a08b688cbe60acc386ed1fbd1b7cbaaf5576b1
commit-date: 2026-05-30
host: aarch64-apple-darwin
release: 1.98.0-nightly
LLVM version: 22.1.6
```
